src/app/beer_garden/db/mongo/models.py

```diff
@@ -401,39 +400,45 @@ class Request(MongoModel, Document):
     logger = logging.getLogger(__name__)
 
     def pre_serialize(self):
+        """Pull any fields out of GridFS"""
         encoding = "utf-8"
-        """If string output was over 16MB it was spilled over to the GridFS storage solution"""
+
         if self.output_gridfs:
-            self.logger.debug("~~~Retrieving output from gridfs")
+            self.logger.debug("Retrieving output from GridFS")
             self.output = self.output_gridfs.read().decode(encoding)
             self.output_gridfs = None
 
         if self.parameters_gridfs:
-            self.logger.debug("~~~Retrieving parameters from gridfs")
+            self.logger.debug("Retrieving parameters from GridFS")
             self.parameters = json.loads(self.parameters_gridfs.read().decode(encoding))
             self.parameters_gridfs = None
 
     def save(self, *args, **kwargs):
+        """Save a request, moving request attributes to GridFS if too big"""
         self.updated_at = datetime.datetime.utcnow()
-        max_size = 15 * 1_000_000
+        # 5MB
+        max_param_size = 5 * 1_000_000
+        max_output_size = 5 * 1_000_000
         encoding = "utf-8"
-        parameter_size = sys.getsizeof(self.parameters)
 
-        if self.output:
-            parameter_size = sys.getsizeof(self.parameters)
-            if parameter_size > max_size:
-                self.logger.info("~~~Parameter size too big, storing in gridfs")
+        def get_size(data):
+            # This func may not work for multi-byte encodings (e.g. utf-16)
+            klen = sum(len(k.encode(encoding)) for k in data.keys())
+            vlen = sum(len(v.encode(encoding)) for v in data.values())
```

Comment:
I haven't fully tested or reviewed this yet, but quick comment here because I immediately ran into a failure on this because this line seems to assume that the values are some encodable type?  Specifically I tasked something that takes a boolean parameter (I think it was "echo") and this blew up.

I think I originally suggested just stringifying the parameters and checking the length.  Ultimately that's how the parameters_gridfs get stored (see the `json.dumps(self.parameters)` line a little below this one. I'm open to the idea of that being problematic for its own reasons.

I'll look at this some more tomorrow, but just wanted to get this comment on here in the meantime.

Reply:
Yeah, following up on this, the stringified json (the results of the json.dumps) seems like the thing you want to check the length of.  So I think maybe what you want to do for both the parameters and the output is to do the `json.dumps` first, then just check the length of that where you're doing the `get_size` call.  So something like:

```python
parameters_json = json.dumps(self.parameters)
output_json = json.dumps(self.output)

if self.parameters:
    if len(parameters_json) > max_param_size:
        ...
        ...
```

Reply:
Unsure of where I got this from, but was under the impression that the Request.parameters could only be type `Dict` with `String`s inside. Most recent commit fixes bools breaking it, as it no longer tries to encode (even though it needs the encoding for the `.put()` method).